### ISSUE TITLE
`libcublas-dev` should depend on `cuda-crt-dev_*`

### ISSUE THREAD

**Issue by adibbley** [conda-forge/libcublas-feedstock#8]
cublas headers depend on cudart headers, which depend on nvcc (crt) headers. We should add these dependencies once those packages are available.

**Comment by bdice** [comment]
When compilers are available, we can also add tests that compile a sample executable (but can't run it without a GPU). Suggesting this, which was proposed by @robertmaynard.

```cpp
#include <cublas_v2.h>

int main() {
    cublasStatus_t stat;
    cublasHandle_t handle;
    stat = cublasCreate(&handle);
    return stat != CUBLAS_STATUS_SUCCESS;
}
```

Then we'd depend on compilers in the test and execute something like `g++ -I... -lcublas test.cpp`.

**Comment by jakirkham** [comment]
Since `{{ compiler("cuda") }}` is going to be needed as a `requirement/build` by any recipe build CUDA packages, this should be picked up via `cuda-nvcc`. If not, we may need to tweak the dependencies of `cuda-nvcc` and/or `cuda-cudart` to make sure that is happening

**Comment by bdice** [comment]
That’s fine if there are downstream recipes, but we still need these requirements on the libcublas recipe if users are building from source.

**Comment by jakirkham** [comment]
cc @billysuh7 (to look into this further)

**Comment by jakirkham** [comment]
Now that the CRT headers are split out of NVCC as of CUDA 12.2 ( https://github.com/conda-forge/cuda-nvcc-feedstock/pull/32 ), this should be doable

It sounds like `cudart` is not needed for cuBLAS. So that doesn't need to be added (confirmed by running `ldd` on `libcublas.so` to check for `libcudart` linkages)

In terms of cuBLAS packages, these make most sense to add to the `-dev` packages

**Comment by bdice** [comment]
@jakirkham Running ldd doesn’t check for dlopen, right? Is there a possibility of libcudart being dlopened, like nvrtc in #9?

**Comment by leofang** [comment]
All math libs must link to cudart statically. It's a requirement.

**Comment by bdice** [comment]
Great, then I agree we can make progress on this when someone has time.

**Comment by billysuh7** [comment]
I will take this. Thanks.

**Comment by jakirkham** [comment]
Should add the above comment ( https://github.com/conda-forge/libcublas-feedstock/issues/8#issuecomment-2259092524 ) was a summary of our discussion in the meeting yesterday ( with feedback from @adibbley and @robertmaynard )

**Comment by jakirkham** [comment]
Retitled the issue to hopefully make this clearer

Thanks Billy! 🙏

**Comment by billysuh7** [comment]
Closing this issue as it was addressed by PR #29
